### PR TITLE
Create independent SetCarriersCommand

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | In the process of uniting all the product commands into one (for its proper fields at least) we realized that `UpdateProductShippingCommand` also handles the carriers relationship The command will soon be removed so we need to handle this association in a dedicated command SetCarriersCommand
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | ~
| Related PRs       | This PR is blocking this one https://github.com/PrestaShop/PrestaShop/pull/30228
| How to test?      | CI tests green should be enough, we'll wait for the PR that finalizes the unified command integration to check for potential regressions (a QA by dev has been done already)
| Possible impacts? | ~

### CODE REVIEW

File: tests/Integration/Behaviour/Features/Context/Domain/Product/UpdateShippingFeatureContext.php [hunk 1]
```diff
@@ -306,23 +357,18 @@ private function setUpdateShippingCommandData(array $data, UpdateProductShipping
...
-            $command->setCarrierReferenceIds($this->getCarrierReferenceIds($data['carriers']));
-            unset($unhandledValues['carriers']);
-        }
-
         return $unhandledValues;
     }
 
     /**
-     * @param string $carrierReferencesInput
+     * @param string[] $carrierReferencesInput
```
Comment (zuk3975) [inline tests/Integration/Behaviour/Features/Context/Domain/Product/UpdateShippingFeatureContext.php]:
```suggestion
     * @param string[] $carrierReferences
```

**Review by zuk3975 — APPROVED** [review]

File: src/PrestaShopBundle/Resources/config/services/adapter/product.yml [hunk 2]
```diff
@@ -225,6 +225,14 @@ services:
       - name: tactician.handler
         command: PrestaShop\PrestaShop\Core\Domain\Product\Command\UpdateProductShippingCommand
 
+  prestashop.adapter.product.command_handler.set_carriers_handler:
```
Comment (yanmakouf) [inline src/PrestaShopBundle/Resources/config/services/adapter/product.yml:228]: Maybe too early for this blocking PR but we should think of using the new standard in the future ;) https://github.com/PrestaShop/ADR/issues/28

**Review by yanmakouf — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]
Looks ok for me

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
> CI tests green should be enough, we'll wait for the PR that finalizes the unified command integration to check for potential regressions (a QA by dev has been done already)

ok, in that case, I'm merging it :)

**Pull request merged**